Integrations/integration-Cymon.yml

```diff
@@ -0,0 +1,543 @@
+            }
+        }
+
+        return context
+
+    def get_domain_report_command():
+        args = demisto.args()
+
+        domain = args.get('domain')
+        full_response = args.get('fullResponse') == 'true'
+
+        domain_report = get_domain_report(domain)
+        domain_full_report = []
+
+        if full_response:
+            domain_full_report = get_domain_full_report(domain)
+
+        if not domain_report.get('reports') or not domain_report.get('total'):
+            return "Domain " + domain + " is not in Cymons's dataset"
+
+        mark_down = create_domain_command_markdown(domain, domain_report.get('total'), domain_report.get('reports'), domain_full_report)
+        context = create_context_domain_command(domain, domain_report.get('reports'))
+
+        return {
+                'Type': entryTypes['note'],
+                'Contents': domain_report,
+                'ContentsFormat': formats['json'],
+                'HumanReadable': mark_down,
+                'EntryContext': context
+            }
+
+    ''' IP COMMAND '''
+    def get_ip_events_sources(ip):
+        url = '{}/{}/{}'.format(SERVER_URL_V1, 'ip', ip)
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        return response.get('sources', None)
+
+    def get_ip_events(ip):
+        url = '{}/{}/{}/{}?limit={}'.format(SERVER_URL_V1, 'ip', ip, 'events', 100)
+        events = {}
+
+        next_link = url
+
+        while next_link is not None:
+            response = http_request('GET', next_link, DEFAULT_HEADERS)
+
+            for event in response.get('results', []):
+                tag = event.get('tag')
+                date = datetime.datetime.strptime(event.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+                if (tag in events and events[tag] < date) or tag not in events:
+                    events.update({tag: date})
+
+            next_link = response.get('next')
+
+        for event in events:
+            events[event] = events[event].strftime("%Y-%m-%d %H:%M:%S")
+
+        return events
+
+    def get_ip_location(ip):
+        url = '{}/{}/{}'.format(SERVER_DASHBOARD_URL_V1, 'geolocation', ip)
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        lon = response.get('longitude', None)
+        lat = response.get('latitude', None)
+
+        if not lon or not lat:
+            return {}
+        else:
+            return {
+                'lon': lon,
+                'lat': lat
+            }
+
+    def get_ip_domains(ip, max_len):
+        url = '{}/{}/{}/{}?limit={}'.format(SERVER_URL_V1, 'ip', ip, 'domains', max_len)
+        domains = []
+
+        response = http_request('GET', url, DEFAULT_HEADERS)
+
+        for domain in response.get('results', []):
+            date = datetime.datetime.strptime(domain.get('updated'), '%Y-%m-%dT%H:%M:%SZ')
+
+            domains.append({'Hostname': domain.get('name'),
+                            'Last Resolved': date.strftime("%Y-%m-%d %H:%M:%S")})
```

Comment:
@MarkShleifer  same as `timestamp`

Reply:
No need here, because the date is validated before